### PR TITLE
feat(radar_object_tracker): add distance and lanelet map based noise filtering for radar tracking

### PR DESCRIPTION
**Pull request by YoshiRi**
## Description

Radar object based tracking has many FPs.
This PR tries to suppress these FPs with following information:

- **distance from self vehicle**
  - Since radar sensor is often set to watch far objects, so we can filter closer objects
- **lanelet map**
  - Lane information is also useful to focus on important object
  - We can filter out objects which is not close to valid lanelet
  - Also, we can filter out the object with high lateral velocity to lane

image

This PR also including refactoring process to put the ros paramter outside the cpp file.

### Test 1: Far bus detection in Odaiba.

Scene image: [TIER IV INTERNAL URL](https://drive.google.com/file/d/14T3RkE6nCvdeyr3cXJF0WtnDOdX6VGWY/view?usp=drive_link).
Rosbag file: See number 4. file in [TIER IV INTERNAL TICKET](https://tier4.atlassian.net/browse/RT1-3592).

image

### Test2: Far train(yurikamome) FP filtering

See [youtube](https://www.youtube.com/watch?v=rllXrP3IAY8).

This proposed filtering can filter out conventional(green) FP tracks and able to generate lower noise radar tracking.

## Interface changes

Nothing

## Effects on system behavior

Nothing

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

- New Feature: Improved radar object tracking by incorporating distance and lanelet map-based noise filtering, reducing false positives.
- Refactor: Moved ROS parameters outside the CPP file for better organization and maintainability.
- Documentation: Updated README.md and models.md to reflect the changes in noise filtering for radar object tracking.
- Bug Fix: Fixed an issue in radar_object_tracker.launch.xml by adding a new parameter for the path to the `radar_object_tracker.param.yaml` file and remapping the `/vector_map` topic.
- Chore: Added a dependency on `lanelet2_extension` in the package manifest file to incorporate functionality from the library into the radar object tracking module.

### CODE REVIEW

File: perception/radar_object_tracker/models.md [hunk 1]
```diff
@@ -74,3 +74,25 @@ v_{k+1} &= v_k \\
...
+
+### minimum range filter
+
+In most cases, Radar sensors are used with other sensors such as LiDAR and Camera, and Radar sensors are used to detect objects far away. So we can filter out objects that are too close to the sensor.
+
+### lanelet based filter
+
+With lanelet map information, We can filter out false positive objects that are not likely important obstacles.
+
+We fukter out objects that satisfy the following conditions:
```
Comment (yukkysaito) [inline perception/radar_object_tracker/models.md:94]:
```suggestion
We filter out objects that satisfy the following conditions:
```

File: perception/radar_object_tracker/src/radar_object_tracker_node/radar_object_tracker_node.cpp [hunk 2]
```diff
@@ -252,6 +415,51 @@ void RadarObjectTrackerNode::checkTrackerLifeCycle(
...
+      // no_closest_lanelet << " velocity direction flag:" << is_velocity_direction_close_to_lanelet
+      // << std::endl;
+      itr = list_tracker.erase(itr);
+      --itr;
+    }
+  }
+}
+
+// remove objects by distance
+void RadarObjectTrackerNode::distanceBasedNoiseFilter(
```
Comment (yukkysaito) [inline perception/radar_object_tracker/src/radar_object_tracker_node/radar_object_tracker_node.cpp:444]: @coderabbitai Would it be easier to understand the method here if we start with the verb?

File: perception/radar_object_tracker/README.md [hunk 3]
```diff
@@ -40,20 +41,29 @@
...
+| `can_assign_matrix`                  | array  |                             | An array of integers used in the data association algorithm                                                     |
+| `max_dist_matrix`                    | array  |                             | An array of doubles used in the data association algorithm                                                      |
+| `max_area_matrix`                    | array  |                             | An array of doubles used in the data association algorithm                                                      |
+| `min_area_matrix`                    | array  |                             | An array of doubles used in the data association algorithm                                                      |
+| `max_rad_matrix`                     | array  |                             | An array of doubles used in the data association algorithm                                                      |
+| `min_iou_matrix`                     | array  |                             | An array of doubles used in the data association algorithm                                                      |
```
[the remark this answers is not in the document]
Comment (yukkysaito) [reply perception/radar_object_tracker/README.md:64]: @coderabbitai Yes. We know

File: perception/radar_object_tracker/models.md [hunk 4]
```diff
@@ -74,3 +74,25 @@
...
+
+- too large lateral distance from lane
+- velocity direction is too different from lane direction
+- too large lateral velocity
```
[the remark this answers is not in the document]
Comment (YoshiRi) [reply perception/radar_object_tracker/models.md]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4935/commits/2f0c534887bb3b5c38b8bd776f4e675dbc81cf34

File: perception/radar_object_tracker/models.md (hunk 1 above)
Comment (YoshiRi) [reply perception/radar_object_tracker/models.md]: Thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/4935/commits/2f0c534887bb3b5c38b8bd776f4e675dbc81cf34.

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**